src/DataSource/DibiFluentDataSource.php

```diff
@@ -60,7 +60,9 @@ public function getCount(): int
 	 */
 	public function getData(): array
 	{
-		return $this->data ?: $this->dataSource->fetchAll();
+		return $this->data !== []
+			? $this->data
+			: $this->dataSource->fetchAll();
```

Comment:
Shouldn't that also write into $this->data, if it's empty?
Maybe property should also be null by default to distinguish between uninitialized and no data states